acracensor/handlers/querycapture_handler.go

```diff
@@ -85,11 +85,15 @@ func NewQueryCaptureHandler(filePath string) (*QueryCaptureHandler, error) {
 				//do nothing. This means that channel has no data to read yet
 			}
 			time.Sleep(handler.serializationTimeout)
-			//timer finished. Close channel to inform that serialization should be performed
 			signalToSerialize <- true
 		}
 	}()
 
+	f, err := os.OpenFile(filePath, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		log.WithError(ErrSingleQueryCaptureError).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorCensorSecurityError)
```

Comment:
what if you can't open file? here we should return error and stop processing. this handler initialized at start of program and should be handled by user before any requests.